### PR TITLE
move x86 tests from Travis to GHA, add aarch64 wheel build to Travis

### PR DESCRIPTION
**Pull request by janaknat**
Move Travis x86 tests to Github Actions. Also, add Travis-CI config to build aarch64 wheels.

### CODE REVIEW

**Comment by janaknat** [comment]
@mpenkov I moved the x86 tests to GHA to preserve Travis credits.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Tests
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    branches: [ develop ]
+  schedule:
+    - cron: '0 0 * * sun,wed'
```
Comment (mpenkov) [inline .github/workflows/tests.yml:8]: Why is this here? These should run on each PR, not on a schedule.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - {name: Linux, python: 3.6, os: ubuntu-16.04, tox: 'flake8,flake8-docs'}
```
Comment (mpenkov) [inline .github/workflows/tests.yml:21]: Should we use a more recent Ubuntu version?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Update pip
+        run: python -m pip install -U pip
+      - name: Install tox, gdb
+        run: |
+          pip install tox
+          sudo apt-get install -y gdb
+      - name: Enable core dumps
+        run: ulimit -c unlimited -S  # enable core dumps
+      - name: Run tox tests
+        run: tox -e ${{ matrix.tox }} -vv
```
Comment (mpenkov) [inline .github/workflows/tests.yml:43]: Is vv necessary here? Tox will spew lots of output that nobody really ever reads.

File: .travis.yml [hunk 4]
```diff
@@ -1,53 +1,53 @@
...
-cache:
-  apt: true
-  directories:
-  - $HOME/.cache/pip
-  - $HOME/.ccache
-  - $HOME/.pip-cache
-dist: trusty
+branches:
+  only:
+    - develop
```
Comment (mpenkov) [inline .travis.yml:3]: What does this file actually do? It looks to me like it will build a wheel on every commit, which isn't what we want, right?

File: .travis.yml [hunk 5]
```diff
@@ -1,53 +1,53 @@
...
-  - ulimit -c unlimited -S  # enable core dumps
-
-
-script: tox -vv
-
-
+    - build_wheel $REPO_DIR $PLAT
+script:
+    - install_run $PLAT
+after_success:
```
Comment (mpenkov) [inline .travis.yml:40]: Why are after_success and after_failure steps exactly the same?

File: .github/workflows/tests.yml (hunk 2 above)
Comment (janaknat) [reply .github/workflows/tests.yml:21]: I followed the existing tests in .travis.yml. I can update to use 'ubuntu-latest' for everything.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (janaknat) [reply .github/workflows/tests.yml:43]: I'm following the existing tests in .travis.yml. I can remove it.

File: .travis.yml (hunk 4 above)
Comment (janaknat) [reply .travis.yml:3]: I guess not. I can build on every release. Is that done through master?

File: .travis.yml (hunk 5 above)
Comment (janaknat) [reply .travis.yml:40]: Should I remove the after_failure? I followed what gensim-wheels was doing.

**Comment by janaknat** [comment]
@mpenkov I've made the changes.

**Comment by mpenkov** [comment]
@janaknat Thank you.

In the future, can you please avoid force-pushing? It makes it more difficult to review your PR as a maintainer.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (mpenkov) [reply .github/workflows/tests.yml:19]: I think 20.04 is the way to go here.

File: .travis.yml (hunk 5 above)
Comment (mpenkov) [reply .travis.yml:40]: If the two are identical, then yes. More specifically:

- remove after_failure
- make after_success another step (the last step in the workflow)
- ensure that all steps in the workflow run, even if preceding steps fail

**Comment by janaknat** [comment]
@mpenkov I've made the changes.

File: .travis.yml (hunk 4 above)
Comment (mpenkov) [reply .travis.yml]: Releases are done on each tag that matches the regex `v\d+\.\d+\.\d+`

Comment (janaknat) [reply .travis.yml]: You can create a cron job from the UI. Can't find any documentation about setting it up through the .travis.yml file. https://docs.travis-ci.com/user/cron-jobs/

File: .travis.yml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 branches:
   only:
-    - master
+    - /v\d+\.\d+\.\d+/
```
Comment (mpenkov) [inline .travis.yml:3]: This needs to match to a tag name, not a branch name.

Comment (janaknat) [reply .travis.yml:3]: Based on https://docs.travis-ci.com/user/customizing-the-build/#safelisting-or-blocklisting-branches I believe having a regex in branches.only will only build tagged commits.

Tagged commit build: https://travis-ci.com/github/janaknat/gensim/builds/214207365

**Comment by janaknat** [comment]
@mpenkov Is there anything in the PR that I should address?

**Comment by janaknat** [comment]
@mpenkov Ping.

**Comment by mpenkov** [comment]
Sorry, I was busy with other projects during the week. Thank you for your patience.

Let's merge this and see how it plays out in the wild.

**Comment by mpenkov** [comment]
Looks like there are some problems, @janaknat please investigate.

https://github.com/RaRe-Technologies/gensim/runs/1796402952?check_suite_focus=true

**Comment by janaknat** [comment]
@mpenkov Are there any caches being used? The errors look like issues with trying to download tox, gdb and the IP not being accessible.

**Comment by mpenkov** [comment]
Not that I'm aware of. The github actions stuff is all very new, from memory, you're one of the first people to contribute to it with gensim.

**Comment by janaknat** [comment]
@mpenkov I've got a PR that solved the issue for me. Looks like the repos are out of date. An apt update was required. https://github.com/RaRe-Technologies/gensim/pull/3035

**Comment by janaknat** [comment]
@mpenkov Any dates on when the aarch64 wheels show up in PyPI?

**Comment by piskvorky** [comment]
For my part, I plan to do an open source sprint in February to finish up my tasks for the [4.0.0 Milestone](https://github.com/RaRe-Technologies/gensim/milestone/3).

Which of course doesn't mean we cannot do another beta / alpha release before that – up to @mpenkov .

**Comment by piskvorky** [comment]
We're planning 4.0.0rc1 (release candidate) this week. And then the "full" 4.0.0 right after :)

**Comment by janaknat** [comment]
That's great. aarch64 is on track to get wheels this time around?

**Comment by piskvorky** [comment]
Hm, I thought that's what this PR was about. @mpenkov will any other steps be necessary to get from here to wheels on PyPI?

**Comment by mpenkov** [comment]
We'll look at it as part of the release process. If there's any additional work required to get the wheels into PyPI once they're built, we'll take care of it then.

**Comment by janaknat** [comment]
Hey @mpenkov @piskvorky I see gensim has a new release. I don't see aarch64 wheels. Any issues?

**Comment by mpenkov** [comment]
I couldn't get the pipeline working when making the release. I'll debug the issue.

Thank you for reminding me.

**Comment by mpenkov** [comment]
@janaknat Please have a look at this: https://travis-ci.org/github/RaRe-Technologies/gensim/jobs/764640506

**Comment by janaknat** [comment]
@mpenkov You need to migrate to Travis-ci.com from travis-ci.org. Arm64 runners are only available in .com.

**Comment by mpenkov** [comment]
OK, so this is related to #3083.
